base/v0_5_exp/translate_test.go

```diff
@@ -1699,6 +1699,293 @@ func TestTranslateKernelArguments(t *testing.T) {
+			assert.Equal(t, test.out, actual, "translation mismatch")
+			expectedReport := strings.ReplaceAll(test.report, "%FilePath%", test.filePath)
+			assert.True(t, strings.HasPrefix(r.String(), expectedReport), "bad report")
+			assert.NoError(t, translations.DebugVerifyCoverage(actual), "incomplete TranslationSet coverage")
+		})
+	}
+}
+
+// TestTranslateSystemdLocalUnits tests translating the butane systemd.units[i].contents_local entries to ignition systemd.units[i].contents entries.
+func TestTranslateSystemdLocalUnits(t *testing.T) {
+	unitDir := t.TempDir()
+	randomDir := t.TempDir()
+	var unitName = "dummy.service"
+	var unitDefinition = "[Service]\nExecStart=/bin/false\n"
+	var unitEmptyFileName = "empty.service"
+	var unitEmptyDefinition = ""
+	var unitNonExistingFileName = "random.service"
+
+	err := ioutil.WriteFile(filepath.Join(unitDir, unitName), []byte(unitDefinition), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+	err = ioutil.WriteFile(filepath.Join(unitDir, unitEmptyFileName), []byte(unitEmptyDefinition), 0644)
+	if err != nil {
+		t.Error(err)
+	}
+
+	tests := []struct {
+		name     string
+		in       Unit
+		out      types.Unit
+		report   string
+		fileDir  string
+		filePath string
+	}{
+		{
+			"empty unit",
+			Unit{},
+			types.Unit{},
+			"",
+			"",
+			"",
+		},
+		{
+			"valid contents",
+			Unit{Contents: &unitDefinition, Name: unitName},
+			types.Unit{Contents: &unitDefinition, Name: unitName},
+			"",
+			"",
+			"",
+		},
+		{
+			"valid contents_local",
+			Unit{ContentsLocal: &unitName, Name: unitName},
+			types.Unit{Contents: &unitDefinition, Name: unitName},
+			"",
+			unitDir,
+			filepath.Join(unitDir, unitName),
+		},
+		{
+			"non existing contents_local file name",
+			Unit{ContentsLocal: &unitNonExistingFileName, Name: unitName},
+			types.Unit{Name: unitName},
+			"error at $.contents: open %FilePath%:",
+			unitDir,
+			filepath.Join(unitDir, unitNonExistingFileName),
+		},
+		{
+			"valid empty contents_local file",
+			Unit{ContentsLocal: &unitEmptyFileName, Name: unitName},
+			types.Unit{Contents: &unitEmptyDefinition, Name: unitName},
+			"",
+			unitDir,
+			filepath.Join(unitDir, unitEmptyFileName),
+		},
+		{
+			"missing embed directory",
+			Unit{ContentsLocal: &unitName, Name: unitName},
+			types.Unit{Name: unitName},
+			"error at $.contents: local file paths are relative to a files directory that must be specified with -d/--files-dir\n",
+			"",
+			"",
+		},
+		{
+			"wrong embed directory",
+			Unit{ContentsLocal: &unitName, Name: unitName},
+			types.Unit{Name: unitName},
+			"error at $.contents: open %FilePath%:",
+			randomDir,
+			filepath.Join(randomDir, unitName),
+		},
+		{
+			"empty dropin unit",
+			Unit{Name: unitName, Dropins: nil},
+			types.Unit{Name: unitName, Dropins: nil},
+			"",
+			"",
+			"",
+		},
+		{
+			"valid dropin contents",
+			Unit{Dropins: []Dropin{{Name: unitName, Contents: &unitDefinition}}, Name: unitName},
```

Comment:
Nit: this test doesn't really care, but actually the dropin name should be something.conf.